### PR TITLE
Added Time graph model and updates to the main class

### PR DESCRIPTION
**Pull request by bhufmann**

### CODE REVIEW

File: tsp/entry_model.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import copy
+from tsp.model_type import ModelType
+from tsp.time_graph_model import TimeGraphEntry
+from tsp.entry import EntryHeader, Entry
+
+HEADER_KEY = "headers"
+ENTRIES_KEY = "entries"
+
+class EntryModel(object):
```
Comment (MatthewKhouzam) [inline tsp/entry_model.py]: tsp/entry_model.py:31:1: E302 expected 2 blank lines, found 1

File: tsp/entry_model.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        '''
+        Array of entry
+        '''
+        self.entries = []
+        if ENTRIES_KEY in params:
+            entries = params.get(ENTRIES_KEY)
+            for entry in entries:
+                if model_type == ModelType.TIME_GRAPH_TREE:
+                    self.entries.append(TimeGraphEntry(entry))
+                else:
```
Comment (MatthewKhouzam) [inline tsp/entry_model.py]: tsp/entry_model.py:54:22: W291 trailing whitespace

File: tsp/entry_model.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            entries = params.get(ENTRIES_KEY)
+            for entry in entries:
+                if model_type == ModelType.TIME_GRAPH_TREE:
+                    self.entries.append(TimeGraphEntry(entry))
+                else:       
+                    self.entries.append(Entry(entry))
+            del params[ENTRIES_KEY]
+
+        '''
+        Store other key/value pairs that are not defined in the TSP in a dictionary
```
Comment (MatthewKhouzam) [inline tsp/entry_model.py]: tsp/entry_model.py:59:80: E501 line too long (83 > 79 characters)

I don't care about this, reporting pycodestyle

File: tsp/entry_model.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        '''
+        self.others = {}
+        if params:
+            self.others = copy.deepcopy(params)
+
```
Comment (MatthewKhouzam) [inline tsp/entry_model.py]: tsp/entry_model.py:64:1: W293 blank line contains whitespace
tsp/entry_model.py:64:1: W391 blank line at end of file

File: tsp/model_type.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+from enum import Enum
+
+class ModelType(Enum):
```
Comment (MatthewKhouzam) [inline tsp/model_type.py]: tsp/model_type.py:24:1: E302 expected 2 blank lines, found 1

File: tsp/time_graph_model.py [hunk 6]
```diff
@@ -0,0 +1,233 @@
+# The MIT License (MIT)
```
Comment (MatthewKhouzam) [inline tsp/time_graph_model.py]: tsp/time_graph_model.py:43:1: E302 expected 2 blank lines, found 0
tsp/time_graph_model.py:58:39: W291 trailing whitespace
tsp/time_graph_model.py:65:37: W291 trailing whitespace
tsp/time_graph_model.py:73:1: W293 blank line contains whitespace
tsp/time_graph_model.py:75:80: E501 line too long (83 > 79 characters)
tsp/time_graph_model.py:81:1: E305 expected 2 blank lines after class or function definition, found 1
tsp/time_graph_model.py:84:1: E302 expected 2 blank lines, found 0
tsp/time_graph_model.py:85:1: W293 blank line contains whitespace
tsp/time_graph_model.py:93:80: E501 line too long (83 > 79 characters)
tsp/time_graph_model.py:99:1: E305 expected 2 blank lines after class or function definition, found 1
tsp/time_graph_model.py:102:1: E302 expected 2 blank lines, found 0
tsp/time_graph_model.py:121:80: E501 line too long (83 > 79 characters)
tsp/time_graph_model.py:127:1: E305 expected 2 blank lines after class or function definition, found 1
tsp/time_graph_model.py:130:1: E302 expected 2 blank lines, found 0
tsp/time_graph_model.py:151:34: W291 trailing whitespace
tsp/time_graph_model.py:165:33: W291 trailing whitespace
tsp/time_graph_model.py:172:34: W291 trailing whitespace
tsp/time_graph_model.py:173:1: W293 blank line contains whitespace
tsp/time_graph_model.py:175:80: E501 line too long (83 > 79 characters)
tsp/time_graph_model.py:181:1: E305 expected 2 blank lines after class or function definition, found 1
tsp/time_graph_model.py:184:1: E302 expected 2 blank lines, found 0
tsp/time_graph_model.py:226:34: W291 trailing whitespace
tsp/time_graph_model.py:229:80: E501 line too long (83 > 79 characters)

File: tsp/entry_model.py (hunk 1 above)
Comment (bhufmann) [reply tsp/entry_model.py]: Done

File: tsp/entry_model.py (hunk 3 above)
Comment (bhufmann) [reply tsp/entry_model.py]: Done

File: tsp/entry_model.py (hunk 4 above)
Comment (bhufmann) [reply tsp/entry_model.py]: Done

File: tsp/model_type.py (hunk 5 above)
Comment (bhufmann) [reply tsp/model_type.py]: Done

File: tsp/time_graph_model.py (hunk 6 above)
Comment (bhufmann) [reply tsp/time_graph_model.py]: Done (I think)

**Comment by bhufmann** [comment]
thanks

**Pull request merged**